plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtDeleteVMSnapshotCommandWrapper.java

```diff
@@ -63,6 +66,14 @@ public Answer execute(final DeleteVMSnapshotCommand cmd, final LibvirtComputingR
             dm.suspend(); // suspend the vm to avoid image corruption
 
             snapshot.delete(0); // only remove this snapshot, not children
+            didDelete = true;
+
+            // Resume the VM
+            dm = libvirtComputingResource.getDomain(conn, vmName);
+            state = dm.getInfo().state;
+            if (state == DomainInfo.DomainState.VIR_DOMAIN_PAUSED) {
+                dm.resume();
+            }
```

Comment:
resume action leads to an implicit assumption which must be guarded during maintenance by future generations: failing resume is concluded by examining a var named `didDelete`. dangerous assumption even with the current logic computing. see below.